### PR TITLE
Added starter blocks

### PR DESCRIPTION
**Pull request by MrNicholson85**
This change allows adds the ability to create Gutenberg blocks using  ACF/Wordplate (ACF/Extend)

### CODE REVIEW

File: App/Blocks/RegisterBlocks.php [hunk 1]
```diff
@@ -40,7 +40,7 @@ public function registerBlocks()
 
         wp_register_style(
             $this->block_key,
-            get_template_directory_uri() . '/build/css/editor.min.css',
+            get_stylesheet_directory_uri() . '/build/css/editor.min.css',
```
Comment (gojsse) [inline App/Blocks/RegisterBlocks.php:43]: Why did you change this function? get_template_directory_uri should be the right way to call this.

Comment (gojsse) [reply App/Blocks/RegisterBlocks.php:43]: And maybe I'm wrong on this... I know that is how it should work with enqueue_style and enqueue_script

Comment (MrNicholson85) [reply App/Blocks/RegisterBlocks.php:43]: So, I looked in to what this code is doing and it's seems to be overwriting the admin editor. With this code in place the editor is full width (which I don't really like lol).

So I'm down to remove it.

Comment (gojsse) [reply App/Blocks/RegisterBlocks.php:43]: You can edit the editor.scss file which includes components/blocks.scss that has the full width property it looks like

File: App/Fields/Common.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+class Common
+{
+   /**
+ * Margin Options Group
+ *
+ * @param  string $label
+ * @return void
+ */
+public static function marginGroup($label = 'Margin',  $defaults = [3, 3])
```
Comment (gojsse) [inline App/Fields/Common.php:24]: Any class methods should be indented

File: App/Blocks/Types/CarouselBlock.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        add_action('acf/init', [$this, 'registerBlockType']);
+    }
+
+    /**
+     * Register block type
+     */
+    public function registerBlockType()
+    {
+        if (!function_exists('acf_register_block_type')) {
```
Comment (gojsse) [inline App/Blocks/Types/CarouselBlock.php:27]: Is this something you will expand on at some point? It doesn't look like it does anything in particular yet.

File: App/Theme/Styles.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+        /**
+     * Output internal css before block markup for custom margin options
+     *
+     * @param  [type] $block_id
+     * @param  [type] $block
+     * @return void
+     */
+    public function outputBlockMarginStyles($block_id, $block)
```
Comment (gojsse) [inline App/Theme/Styles.php:30]: Thoughts on this file existing under the blocks folder? I could go either way.

File: components/blocks/carousel/block.json [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{
+    "name": "carousel",
+    "title": "Carousel",
+    "description": "A custom Carousel block.",
+    "style": [ "file:/test.css" ],
```
Comment (gojsse) [inline components/blocks/carousel/block.json:5]: Does test.css exist or should we point it to a different file name?

Comment (gojsse) [reply components/blocks/carousel/block.json:5]: I almost wonder if we should have a file name standard for these block css files? Maybe just styles.css?

File: components/blocks/carousel/carousel.php [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    $className .= ' ' . $block['className'];
+}
+if( !empty($block['align']) ) {
+    $className .= ' align' . $block['align'];
+}
+
+// Load values and assign defaults.
+$carousel = get_field('carousel');
+
+do_action('mc/styles/margin', $block_id, $block);
```
Comment (gojsse) [inline components/blocks/carousel/carousel.php:34]: It might benefit to expand this namespace to something like `mc/styles/blocks/margin` to show it's block-related only?

File: components/blocks/testimonial/testimonial.php [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    $className .= ' ' . $block['className'];
+}
+if( !empty($block['align']) ) {
+    $className .= ' align' . $block['align'];
+}
+
+// Load values and assign defaults.
+$image            = get_field('image');
+$background_color = get_field('background-color');
+$text_color       = get_field('text-color');
```
Comment (gojsse) [inline components/blocks/testimonial/testimonial.php:33]: We might have to figure out a new utility class/method to get these values. On some bigger sites when using a lot of get_field()s it can bog them down b/c they are separate db calls for each.

File: App/Blocks/Types/CarouselBlock.php (hunk 3 above)
Comment (MrNicholson85) [reply App/Blocks/Types/CarouselBlock.php:27]: Yea, I should get rid of those. they was used in place of the Block.json file. but we now have that working now.

File: components/blocks/testimonial/testimonial.php [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        <?php 
+            printf(
+                '<img src="%1$s">',
+                $image['url']
+            )
+        ?>
+    </div>
+    <style type="text/css">
+        #<?php echo $id; ?> {
+            background: <?php echo $background_color; ?>;
```
Comment (gojsse) [inline components/blocks/testimonial/testimonial.php:86]: do these need to be escaped?

File: components/blocks/carousel/block.json (hunk 5 above)
Comment (MrNicholson85) [reply components/blocks/carousel/block.json:5]: I'm currently working on that as I'm trying to get the admin blocks to inherit the styles from the front-end blocks.
